src/lib/o1trace/thread.ml

```diff
@@ -87,7 +88,7 @@ module Fiber = struct
         fiber
     | None ->
         let thread = register name in
-        let fiber = { id = !next_id; parent; thread } in
+        let fiber = { id = !next_id; parent; thread; key } in
```

Comment:
Should we reconstruct the key on-demand instead of storing it? With this implementation, each fiber will have a unique allocation of a `string list` (without structural sharing). The key can be created from any given fiber by traversing the path up to the root fiber.

Reply:
Yes, we should.